service/matching/taskReader.go

```diff
@@ -149,6 +154,8 @@ func (tr *taskReader) Signal() {
 }
 
 func (tr *taskReader) dispatchBufferedTasks(isolationGroup string) {
+	tr.stopWg.Add(1)
+	defer tr.stopWg.Done()
```

Comment:
Looks like a generic approach for stopping. I wonder if it's possible to have some OOP around this.

Reply:
OOP in Go is hard, I'm not aware of generic solutions on this.